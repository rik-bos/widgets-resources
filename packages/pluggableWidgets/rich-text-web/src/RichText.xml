<?xml version="1.0" encoding="utf-8"?>
<widget id="com.mendix.widget.custom.richtext.RichText" needsEntityContext="true" pluginWidget="true" offlineCapable="true" xmlns="http://www.mendix.com/widget/1.0/"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.mendix.com/widget/1.0/ ../../../../node_modules/mendix/custom_widget.xsd">
    <name>Rich Text</name>
    <description>Rich inline or toolbar text editing</description>
    <icon>
        iVBORw0KGgoAAAANSUhEUgAAAEAAAABACAYAAACqaXHeAAAAAXNSR0IArs4c6QAAA2tJREFUeAHtW72OEzEQ3g1IQJn+gI4WpGsQT0AaREESeIGrrsxrnES9DbRJQDrRHC0010JDjY68ANLBQZPc571dnT32xo4zXhzWadZjz+839uyPnSxLv4RAQqDLCOQ0+OFweAN9h3meP8FVtG2/89Vq9WY+n3+yMXKOj8fjwXK5LKBzz1HvotfrHUyn0xOZ/6ZMVO1DXF8jKMNQY9dLAPcAIJw1cjAPbBi8sL5XydyVXenJhGhXmafdNvoWGPZtTMzjrpmXzWoyGgDgdpn2stKyDeBMujS+2DpMS0DxEYEdoeNU6bwiJlgmj+t+tO9jXT6s6dBXTGfFBJafVs8EA5bm2rVsBQA6Tmez2XvFGojRaPSK9B1Rp8h4lOROTltOJDsPgMsScAIcteI7GH86MTMwoeaw1Bs2ABDTxFQrGGI1qrAVN6OQodMFgImh4GXyHcCgt/UuX0CsAMQWKDeypiJ47mMEQP3ykdtCZuEhq8loAOCF4S0U/91Q+bd+v/95Q5mt2MWLDRRoAa1RWr4M0fGmp6d7YNyXH2+R4XeyMMZeCFpkXgRfFMVvQQ8+/Blky1WBxy/tuVuM+/7g6CLr5Qcnz24rb3O1PloDmp4Ma/76aqwB1VvdWc0krjAgk1ljxQ8QvDAsAM2hG03lbU5xyoPQloCHDkWEO/Oy8hC62QGQHd6FtnEJcDr+8fkdY515enyBhF7/XPmuJXhanZ8B7ACU1bpKjtz2zZesQ2776qNy7ACIW5VwtHQWbWpwY5pbH3GAvQZU92m2WxW3PhJ/xj8DqIXI6QRA5AkK7l6aAcEhjtxAmgGRJyi4e2kGBIc4cgNpBkSeoODupRkQHOLIDaQZEHmCgrvX+gwQ+waD44sfNDLRV+4p0IHAdOsANG2alJ+8r777Bw5ZVd8+AKr9f061D0D1jY9GzvYNkSq20LCr/rhOYKpa+SkuP7UZsMUJTP4o12jk8lMDADZ9dnV9ZNaE5zTkY1OTYfssjmM0X5zcZmLCtjyLJisATfvsdD8eDrGc2vKNytVPqt+0BCjPf013HgDrEnBNP87sPHLl5eDDXYCl5rABgH9ifOUIzFUHPbLjKkf50hKgiFCaVns6Hgvt62fnZ4AJgIVHVn1kPMwoIj42NRkNAK4TmIqrAYhd8TNA6EllQiAhwIjAJSEj9lh5/+kBAAAAAElFTkSuQmCC
    </icon>
    <properties>
        <propertyGroup caption="General">
            <propertyGroup caption="General">
                <property key="stringAttribute" type="attribute">
                    <caption>Value attribute</caption>
                    <description>Rich text HTML content</description>
                    <attributeTypes>
                        <attributeType name="String"/>
                    </attributeTypes>
                </property>
                <property key="sanitizeContent" type="boolean" defaultValue="true">
                    <caption>Sanitize content</caption>
                    <description>The strict sanitize rules will remove all unsupported HTML tags and attributes before displaying.WARNING disabling this option is a security risk for XSS attacks
                    </description>
                </property>
                <property key="editorType" type="enumeration" defaultValue="classic">
                    <caption>Editor theme</caption>
                    <description>Display the editor formatting options in a toolbar or a bubble when text is selected</description>
                    <enumerationValues>
                        <enumerationValue key="classic">Classic</enumerationValue>
                        <enumerationValue key="inline">Inline</enumerationValue>
                    </enumerationValues>
                </property>
                <systemProperty key="Editability"/>
                <property key="readOnlyStyle" type="enumeration" defaultValue="text">
                    <caption>Read-only style</caption>
                    <description/>
                    <enumerationValues>
                        <enumerationValue key="text">Rich text</enumerationValue>
                        <enumerationValue key="bordered">Bordered</enumerationValue>
                        <enumerationValue key="borderedToolbar">Bordered toolbar</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="advancedMode" type="boolean" defaultValue="false">
                    <caption>Enable advanced options</caption>
                    <description/>
                </property>
            </propertyGroup>
        </propertyGroup>
        <propertyGroup caption="Toolbars">
            <propertyGroup caption="Toolbars">
                <property key="preset" type="enumeration" defaultValue="basic">
                    <caption>Presets</caption>
                    <description/>
                    <enumerationValues>
                        <enumerationValue key="basic">Basic</enumerationValue>
                        <enumerationValue key="standard">Standard</enumerationValue>
                        <enumerationValue key="full">Full</enumerationValue>
                        <enumerationValue key="custom">Custom</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="toolbarConfig" type="enumeration" defaultValue="basic">
                    <caption>Toolbar group</caption>
                    <description/>
                    <enumerationValues>
                        <enumerationValue key="basic">Basic</enumerationValue>
                        <enumerationValue key="advanced">Advanced</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="documentGroup" type="enumeration" defaultValue="yes">
                    <caption>Document Group</caption>
                    <description/>
                    <enumerationValues>
                        <enumerationValue key="yes">Yes</enumerationValue>
                        <enumerationValue key="no">No</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="clipboardGroup" type="enumeration" defaultValue="yes">
                    <caption>Clipboard Group</caption>
                    <description/>
                    <enumerationValues>
                        <enumerationValue key="yes">Yes</enumerationValue>
                        <enumerationValue key="no">No</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="editingGroup" type="enumeration" defaultValue="yes">
                    <caption>Editing Group</caption>
                    <description/>
                    <enumerationValues>
                        <enumerationValue key="yes">Yes</enumerationValue>
                        <enumerationValue key="no">No</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="formsGroup" type="enumeration" defaultValue="yes">
                    <caption>Forms group</caption>
                    <description/>
                    <enumerationValues>
                        <enumerationValue key="yes">Yes</enumerationValue>
                        <enumerationValue key="no">No</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="separatorGroup" type="enumeration" defaultValue="yes">
                    <caption>Separator group</caption>
                    <description/>
                    <enumerationValues>
                        <enumerationValue key="yes">Yes</enumerationValue>
                        <enumerationValue key="no">No</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="basicStylesGroup" type="enumeration" defaultValue="yes">
                    <caption>Basic style group</caption>
                    <description/>
                    <enumerationValues>
                        <enumerationValue key="yes">Yes</enumerationValue>
                        <enumerationValue key="no">No</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="paragraphGroup" type="enumeration" defaultValue="yes">
                    <caption>Paragraph group</caption>
                    <description/>
                    <enumerationValues>
                        <enumerationValue key="yes">Yes</enumerationValue>
                        <enumerationValue key="no">No</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="linksGroup" type="enumeration" defaultValue="yes">
                    <caption>Links group</caption>
                    <description/>
                    <enumerationValues>
                        <enumerationValue key="yes">Yes</enumerationValue>
                        <enumerationValue key="no">No</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="separator2Group" type="enumeration" defaultValue="yes">
                    <caption>Separator 2 group</caption>
                    <description/>
                    <enumerationValues>
                        <enumerationValue key="yes">Yes</enumerationValue>
                        <enumerationValue key="no">No</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="stylesGroup" type="enumeration" defaultValue="yes">
                    <caption>Styles group</caption>
                    <description/>
                    <enumerationValues>
                        <enumerationValue key="yes">Yes</enumerationValue>
                        <enumerationValue key="no">No</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="colorsGroup" type="enumeration" defaultValue="yes">
                    <caption>Colors group</caption>
                    <description/>
                    <enumerationValues>
                        <enumerationValue key="yes">Yes</enumerationValue>
                        <enumerationValue key="no">No</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="toolsGroup" type="enumeration" defaultValue="yes">
                    <caption>Tools group</caption>
                    <description/>
                    <enumerationValues>
                        <enumerationValue key="yes">Yes</enumerationValue>
                        <enumerationValue key="no">No</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="othersGroup" type="enumeration" defaultValue="yes">
                    <caption>Others group</caption>
                    <description/>
                    <enumerationValues>
                        <enumerationValue key="yes">Yes</enumerationValue>
                        <enumerationValue key="no">No</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="advancedGroup" type="string" required="false" multiline="true">
                    <caption>Advanced groups</caption>
                    <description/>
                </property>
            </propertyGroup>
        </propertyGroup>
        <propertyGroup caption="Dimensions">
            <propertyGroup caption="Dimensions">
                <property key="widthUnit" type="enumeration" defaultValue="percentage">
                    <caption>Width unit</caption>
                    <description>Percentage: portion of parent size. Pixels: absolute amount of pixels.</description>
                    <enumerationValues>
                        <enumerationValue key="percentage">Percentage</enumerationValue>
                        <enumerationValue key="pixels">Pixels</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="width" type="integer" defaultValue="100">
                    <caption>Width</caption>
                    <description/>
                </property>
                <property key="heightUnit" type="enumeration" defaultValue="percentageOfWidth">
                    <caption>Height unit</caption>
                    <description/>
                    <enumerationValues>
                        <enumerationValue key="percentageOfWidth">Percentage of width</enumerationValue>
                        <enumerationValue key="pixels">Pixels</enumerationValue>
                        <enumerationValue key="percentageOfParent">Percentage of parent</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="height" type="integer" defaultValue="75">
                    <caption>Height</caption>
                    <description/>
                </property>
            </propertyGroup>
        </propertyGroup>
        <propertyGroup caption="Events">
            <property key="onKeyPress" type="action">
                <caption>On key press</caption>
                <description/>
            </property>
            <property key="onChange" type="action">
                <caption>On change</caption>
                <description/>
            </property>
            <property key="enterMode" type="enumeration" defaultValue="paragraph">
                <caption>Enter mode</caption>
                <description/>
                <enumerationValues>
                    <enumerationValue key="paragraph">Paragraph</enumerationValue>
                    <enumerationValue key="breakLines">Break lines</enumerationValue>
                    <enumerationValue key="blocks">Blocks</enumerationValue>
                </enumerationValues>
            </property>
            <property key="shiftEnterMode" type="enumeration" defaultValue="paragraph">
                <caption>Shift enter mode</caption>
                <description/>
                <enumerationValues>
                    <enumerationValue key="paragraph">Paragraph</enumerationValue>
                    <enumerationValue key="breakLines">Break lines</enumerationValue>
                    <enumerationValue key="blocks">Blocks</enumerationValue>
                </enumerationValues>
            </property>
            <property key="autoParagraph" type="enumeration" defaultValue="yes">
                <caption>Auto paragraph</caption>
                <description/>
                <enumerationValues>
                    <enumerationValue key="yes">Yes</enumerationValue>
                    <enumerationValue key="no">No</enumerationValue>
                </enumerationValues>
            </property>
            <property key="spellChecker" type="enumeration" defaultValue="yes">
                <caption>Spell checker</caption>
                <description/>
                <enumerationValues>
                    <enumerationValue key="yes">Yes</enumerationValue>
                    <enumerationValue key="no">No</enumerationValue>
                </enumerationValues>
            </property>
        </propertyGroup>
        <propertyGroup caption="Advanced">
            <property key="codeHighlight" type="enumeration" defaultValue="no">
                <caption>Enable code highlight</caption>
                <description/>
                <enumerationValues>
                    <enumerationValue key="yes">Yes</enumerationValue>
                    <enumerationValue key="no">No</enumerationValue>
                </enumerationValues>
            </property>
            <property key="wordCount" type="enumeration" defaultValue="no">
                <caption>Word count</caption>
                <description/>
                <enumerationValues>
                    <enumerationValue key="yes">Yes</enumerationValue>
                    <enumerationValue key="no">No</enumerationValue>
                </enumerationValues>
            </property>
            <property key="maxChars" type="integer" defaultValue="0">
                <caption>Maximum characters</caption>
                <description/>
            </property>
            <property key="advancedContentFilter" type="enumeration" defaultValue="auto">
                <caption>Advanced content filter</caption>
                <description/>
                <enumerationValues>
                    <enumerationValue key="auto">Auto</enumerationValue>
                    <enumerationValue key="custom">Custom</enumerationValue>
                </enumerationValues>
            </property>
            <property key="allowedContent" type="string" required="false" multiline="true">
                <caption>Allowed content</caption>
                <description/>
            </property>
            <property key="disallowedContent" type="string" required="false" multiline="true">
                <caption>Disallowed content</caption>
                <description/>
            </property>

        </propertyGroup>
    </properties>
</widget>
