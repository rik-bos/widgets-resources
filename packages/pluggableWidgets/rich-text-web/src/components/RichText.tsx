import { createElement, ReactElement, useState, useMemo, useEffect } from "react";
import { CKEditorHookProps, CKEditorType, CKEditorConfig, CKEditorEventAction } from "ckeditor4-react";
import { getDimensions, Dimensions } from "@mendix/piw-utils-internal";
import { defineEnterMode, getPreset, addPlugin } from "../utils/ckeditorConfigs";
import sanitizeHtml from "sanitize-html";
import classNames from "classnames";
import {
    ReadOnlyStyleEnum,
    PresetEnum,
    ToolbarConfigEnum,
    EnterModeEnum,
    ShiftEnterModeEnum,
    AdvancedConfigType
} from "../../typings/RichTextProps";
import { MainEditor } from "./MainEditor";
import { SET_ADVANCED } from "../utils/ckeditorPresets";

export interface RichTextProps {
    autoParagraph: boolean;
    name: string;
    class: string;
    readOnly: boolean;
    spellChecker: boolean;
    sanitizeContent?: boolean;
    value: string;
    advancedConfig: AdvancedConfigType[] | null;
    toolbarGroup: any;
    plugins?: string[];
    readOnlyStyle: ReadOnlyStyleEnum;
    toolbarConfig: ToolbarConfigEnum;
    enterMode?: EnterModeEnum;
    shiftEnterMode?: ShiftEnterModeEnum;
    preset: PresetEnum;
    editorType: CKEditorType;
    dimensions?: Dimensions;
    advancedContentFilter?: { allowedContent: string; disallowedContent: string } | null;
    onValueChange?: (value: string) => void;
    onKeyChange?: () => void;
    onKeyPress?: () => void;
}

export const RichTextEditor = (props: RichTextProps): ReactElement => {
    const {
        editorType,
        preset,
        plugins,
        enterMode,
        shiftEnterMode,
        value,
        readOnly,
        readOnlyStyle,
        advancedContentFilter
    } = props;
    const [element, setElement] = useState<HTMLElement | null>(null);
    const { width, height } = props.dimensions
        ? getDimensions({ ...props.dimensions })
        : {
              width: "100%",
              height: "100%"
          };
    const [ckeditorConfig, setCkeditorConfig] = useState<CKEditorHookProps<any>>({
        element,
        // editorUrl: "/widgets/ckeditor/ckeditor.js",
        type: editorType,
        config: {
            autoGrow_minHeight: 300,
            toolbarCanCollapse: true,
            autoGrow_onStartup: true,
            width,
            height,
            enterMode: defineEnterMode(enterMode || ""),
            shiftEnterMode: defineEnterMode(shiftEnterMode || ""),
            autoParagraph: props.autoParagraph,
            disableNativeSpellChecker: !props.spellChecker
        },
        initContent: value,
        dispatchEvent: ({ type, payload }) => {
            if (type === CKEditorEventAction.change) {
                const value = payload.editor.getData() as string;
                if (props.onKeyChange) {
                    props.onKeyChange();
                }
                if (props.onKeyPress) {
                    props.onKeyPress();
                }
                if (props?.onValueChange) {
                    const content = props.sanitizeContent ? sanitizeHtml(value) : value;
                    props?.onValueChange(content);
                }
            }
        },
        subscribeTo: ["change"]
    });
    const editorPreset: CKEditorConfig | null = getPreset(preset);
    const key = useMemo(() => Date.now(), [ckeditorConfig]);
    const setToolbar = (): CKEditorConfig => {
        if (!editorPreset) {
            const { toolbarGroup, toolbarConfig } = props;
            return SET_ADVANCED(toolbarGroup, toolbarConfig === "basic");
        } else {
            return editorPreset;
        }
    };
    useEffect(() => {
        const config = setToolbar();
        if (plugins?.length) {
            plugins.forEach(plugin => addPlugin(plugin, config));
        }
        if (advancedContentFilter) {
            config.allowedContent = advancedContentFilter.allowedContent;
            config.disallowedContent = advancedContentFilter.disallowedContent;
        }

        setCkeditorConfig({
            ...ckeditorConfig,
            initContent: value,
            element,
            config: {
                ...ckeditorConfig.config,
                ...config
            }
        });
    }, [props]);
    return (
        <div
            className={classNames(props.class, "widget-rich-text", `${readOnly ? `editor-${readOnlyStyle}` : ""}`)}
            style={{ width, height }}
        >
            <div ref={setElement} />
            <MainEditor config={ckeditorConfig} key={key} />
        </div>
    );
};
