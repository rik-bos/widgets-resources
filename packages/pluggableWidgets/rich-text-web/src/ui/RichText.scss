@use "../../../node_modules/ckeditor4/contents.css";
@use "../../../node_modules/ckeditor4/skins/moono-lisa/editor.css";
@use "../../../node_modules/ckeditor4/plugins/tableselection/styles/tableselection.css";
@use "../../../node_modules/ckeditor4/plugins/dialog/styles/dialog.css" as dialog;
@use "../../../node_modules/ckeditor4/plugins/scayt/dialogs/dialog.css" as scaytDialog;
@use "../../../node_modules/ckeditor4/plugins/scayt/skins/moono-lisa/scayt.css";

.widget-rich-text {
    .cke,
    .cke_chrome {
        visibility: visible;
    }

    &.editor-text {
        .cke {
            border: none;
        }

        .cke_top,
        .cke_bottom {
            background: inherit;
            display: none;
        }
    }

    &.editor-bordered {
        .cke {
            border: 1px solid #d1d1d1;
        }

        .cke_top,
        .cke_bottom {
            background: inherit;
            display: none;
        }
    }

    &.editor-borderedToolbar {
    }
}
