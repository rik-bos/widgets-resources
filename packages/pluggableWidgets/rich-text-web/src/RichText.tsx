import { ReactNode, createElement, useCallback } from "react";
import { RichTextEditor as RichTextComponent } from "./components/RichText";
import { RichTextContainerProps } from "../typings/RichTextProps";
import { GroupType } from "./utils/ckeditorPresets";
import { debounce } from "@mendix/piw-utils-internal";
import "./ui/RichText.scss";

export default function RichText(props: RichTextContainerProps): ReactNode {
    const onChangeFn = useCallback(
        debounce((value: string) => props.stringAttribute.setValue(value), 500),
        [props.stringAttribute]
    );
    const toolbarGroup = (): string[] | undefined => {
        const groupKeys = Object.keys(props).filter((key: string) => (key.includes("Group") ? key : null));
        const { toolbarConfig, advancedConfig } = props;
        if (toolbarConfig === "basic") {
            return groupKeys
                .filter((groupName: GroupType) => props[groupName])
                .map((groupName: GroupType) =>
                    groupName.includes("separator") ? "/" : groupName.replace("Group", "").toLowerCase()
                );
        } else if (toolbarConfig === "advanced") {
            return advancedConfig.map(group => (group.ctItemType !== "seperator" ? group.ctItemType : "-"));
        }
    };
    const plugins = [];
    if (props.advancedMode) {
        if (props.codeHighlight) {
            plugins.push("codesnippet");
        }
    }
    return (
        <RichTextComponent
            autoParagraph={props.autoParagraph}
            advancedConfig={props.advancedConfig}
            advancedContentFilter={
                props.advancedContentFilter === "custom"
                    ? {
                          allowedContent: props.allowedContent,
                          disallowedContent: props.disallowedContent
                      }
                    : null
            }
            sanitizeContent={props.sanitizeContent}
            name={props.name}
            class={props.class}
            editorType={props.editorType}
            readOnlyStyle={props.readOnlyStyle}
            readOnly={props.stringAttribute.readOnly}
            preset={props.preset}
            enterMode={props.enterMode}
            shiftEnterMode={props.shiftEnterMode}
            spellChecker={props.spellChecker}
            toolbarConfig={props.toolbarConfig}
            toolbarGroup={toolbarGroup()}
            plugins={plugins}
            value={props.stringAttribute.value as string}
            onValueChange={onChangeFn}
            onKeyPress={props.onKeyPress?.execute}
            onKeyChange={props.onChange?.execute}
            dimensions={{
                width: props.width,
                widthUnit: props.widthUnit,
                height: props.height,
                heightUnit: props.heightUnit
            }}
        />
    );
}
