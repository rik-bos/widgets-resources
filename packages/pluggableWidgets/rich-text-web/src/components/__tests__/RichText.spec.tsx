import { createElement } from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { RichTextEditor, RichTextProps } from "../RichText";

describe("RichText", () => {
    let defaultRichTextProps: RichTextProps;

    beforeEach(() => {
        defaultRichTextProps = {
            advancedConfig: null,
            dimensions: {
                width: 100,
                height: 100,
                heightUnit: "percentageOfWidth",
                widthUnit: "percentage"
            },
            plugins: [],
            value: "Simple Text",
            sanitizeContent: false,
            advancedContentFilter: null,
            readOnly: false,
            name: "RichText",
            preset: "basic",
            toolbarGroup: [],
            spellChecker: false,
            class: "",
            readOnlyStyle: "text",
            autoParagraph: false,
            editorType: "classic",
            enterMode: "paragraph",
            shiftEnterMode: "paragraph",
            toolbarConfig: "basic"
        };
    });
    it("render DOM structure", () => {
        const { asFragment } = render(<RichTextEditor {...defaultRichTextProps} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
