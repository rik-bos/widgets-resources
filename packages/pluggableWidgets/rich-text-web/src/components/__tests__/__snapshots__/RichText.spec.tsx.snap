// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichText render DOM structure 1`] = `
<DocumentFragment>
  <div
    class="widget-rich-text"
    style="width: 100%; height: auto;"
  >
    <div />
  </div>
</DocumentFragment>
`;
